Core/Src/main.c

```diff
@@ -87,17 +92,47 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
   /* USER CODE END 2 */
 
   /* Infinite loop */
+
+  // Starts PWM
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  uint8_t DataReceived[DATASIZE] = {0x00, 0x80, 0x00};
+  uint8_t DataTransmitted[DATASIZE] = {0x00};
+
+
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  // Sets CS line to low
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	  // Uses appropriate SPI protocol to transmit and receive ADC values
+	  HAL_SPI_TransmitReceive(&hspi1, DataTransmitted, DataReceived, sizeof(DataTransmitted), 24);
     /* USER CODE END WHILE */
 
+	  // Delay between processes to ensure that MCU doesnt overload
+	  HAL_Delay(10);
+
     /* USER CODE BEGIN 3 */
+
+	  // Sets CS line back to high
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+	  // Taking the 10 LSB's (B1 to B9) to read from
+	  uint16_t adc_value = ((DataReceived[1] << 8) | DataReceived[2]);
+
+	  // Converting ADC value to number of counts
+	  uint16_t On_Counts = ((DUTY_CYCLE_RANGE * ( adc_value / MAX_ADC))+ DUTY_CYCLE_RANGE);
```

Comment:
Good job. You get the right equation. But I can't confirm the correctness of the result because I can't see your pwm timer configuration. If you are having trouble uploading the `tim.c` file, you can also message me your config in the BootCamp thread. One thing I will have to remain you tho, in this equation, adc_value is not a floating point value. This will lead to `adc_value / MAX_ADC` always result 0 or 1 depending on the compile and eventually result in the wrong `On_Counts`